toml-to-ical/src/tests.rs

```diff
@@ -347,3 +347,45 @@ fn parse_tz_offset() {
     assert!(toml::from_str::<Harness>("offset = \"+450000\"").is_err());
     assert!(toml::from_str::<Harness>("offset = \"-10\"").is_err());
 }
+
+// Check that recurrences must refer to events that exist.
+#[test]
+fn validation_recurrence_does_not_exist() {
+    let cal = Calendar {
+        events: vec![Event {
+            end: Some(Default::default()),
+            recurrence_id: Some(Default::default()),
+            ..Default::default()
+        }],
+        ..Default::default()
+    };
+    assert_eq!(
+        cal.validate().unwrap_err(),
+        ValidationError::RecurrenceDoesNotExist("".to_string())
+    );
+}
+
+// Check that specific recurrences do not themselves have recurrence rules.
+#[test]
+fn validation_recurrence_cannot_have_rules() {
+    let cal = Calendar {
+        events: vec![
+            Event {
+                end: Some(Default::default()),
+                recurrence_rules: RecurrenceRules(vec![Default::default()]),
+                ..Default::default()
+            },
+            Event {
+                end: Some(Default::default()),
+                recurrence_id: Some(Default::default()),
+                recurrence_rules: RecurrenceRules(vec![Default::default()]),
+                ..Default::default()
+            },
+        ],
+        ..Default::default()
+    };
+    assert_eq!(
+        cal.validate().unwrap_err(),
+        ValidationError::RecurrenceCannotHaveRecurrenceRules("".to_string())
+    );
+}
```

Comment:
Missing test for `MismatchedRecurrenceDateTypes`?

Reply:
Fixed!